airflow/providers/google/cloud/operators/cloud_build.py

```diff
@@ -910,8 +910,9 @@ class BuildProcessor:
 
     def __init__(self, build: Union[Dict, Build]) -> None:
         if isinstance(build, Build):
-            self.build = Build(build)
-        self.build = deepcopy(build)
+            self.build: Union[Dict, Build] = Build(build)
+        else:
+            self.build = cast(Dict, deepcopy(build))
```

Comment:
This cast seems redundant?

Reply:
t complained without it - but I can double check it. I think mypy was not able to infer the types of deepcopy from an union. But let's hold off with this one - I do not really like the `# type: ignore` where Google libraries are not following strict-optional and I think I have found a way to solve it (especially that we have many more of similar cases in dataproc and biquery so if we could avoid the ignores in those cases we could fix literally hundred of errors.

Reply:
Strange. It stopped complaining :).

Reply:
But I used the "mypy configuration" trick to enable implicit optionals for the google libraries too, so this PR will get WAY smaller and nicer.